lib/miq_expression/field.rb

```diff
@@ -17,6 +17,10 @@ def self.parse!(field)
     parse(field) or raise ParseError, field
   end
 
+  def self.filter_has_many(cols)
+    cols ? cols.select { |_, column| parse(column).try(:plural?) } : []
```

Comment:
come to think of it, if you move this method to the controller, then you'll get away with only 2 PRs. (one here and one in controller repo)